src/internal/cache-client.ts

```diff
@@ -49,42 +50,49 @@ export class CacheClient {
    * @param {MomentoCacheProps} props
    */
   constructor(props: MomentoCacheProps) {
+    this.configuration = props.configuration;
+    this.authProvider = props.authProvider;
     this.logger = getLogger(this);
-    this.validateRequestTimeout(props.requestTimeoutMs);
+    const grpcConfig = this.configuration
+      .getTransportStrategy()
+      .getGrpcConfig();
+
+    this.requestTimeoutMs =
+      grpcConfig.getDeadlineMilliseconds() ||
+      CacheClient.DEFAULT_REQUEST_TIMEOUT_MS;
```

Comment:
Should grpc defaults live as constants inside the cache client?

Should users of IGrpcConfiguration have to worry about providing a default if getDeadlineMilliseconds is 0?

Should IGrpcConfiguration allow getDeadlineMilliseconds to be 0?

Reply:
> Should grpc defaults live as constants inside the cache client?
> 
> Should users of IGrpcConfiguration have to worry about providing a default if getDeadlineMilliseconds is 0?
> 
> Should IGrpcConfiguration allow getDeadlineMilliseconds to be 0?

These are all fair callouts.  I need to look at some of the other SDKs to see what we're doing for the zero case.  And if the semantics of JS/TS are that `(0 || x) == x` (i.e., 0 is not truth-y) then I think we probably need to make that field nullable and do a better check on it.  I will add that to my list.